eos_data_distribution/ndn/manifest.py

```diff
@@ -32,19 +32,27 @@
 # the manifest producer is an http Producer that answers on a different name
 class Producer(chunks.Producer):
 
-    def __init__(self, name, url, session=None, *args, **kwargs):
+    def __init__(self, name, url, session=None, onHeaders=None, *args, **kwargs):
         self._getter = http.Getter(
-            url, onData=self._send_finish, session=session)
-        self._last_modified = http.get_last_modified(self._getter._headers)
-        if not self._last_modified:
-            raise ValueError("Could not get Last-Modified")
-
-        # XXX -- we mangle the name in the constructor, this is slow
-        self._qualified_name = Name(name).append(self._last_modified)
+            url, session=session,
+            onData=self._send_finish, onHeaders=lambda headers: self._start(headers))
```

Comment:
I think you could replace `lambda headers: self._start(headers)` with just `self._start`.